src/Altinn.App.Core/Internal/Validation/ValidationService.cs

```diff
@@ -163,6 +169,35 @@ public async Task<List<ValidationSourcePair>> ValidateIncrementalFormData(
         return lists.OfType<ValidationSourcePair>().ToList();
     }
 
+    private async Task<List<ValidationIssue>> ValidationIssuesWithDescription(
+        IEnumerable<ValidationIssue> issues,
+        string? language
+    )
+    {
+        if (!issues.Any(i => String.IsNullOrEmpty(i.Description) && !String.IsNullOrEmpty(i.CustomTextKey)))
+        {
+            return issues.ToList();
+        }
+
+        var issuesWithDescription = await Task.WhenAll(
+            issues.Select(async issue => new ValidationIssue
+            {
+                Severity = issue.Severity,
+                DataElementId = issue.DataElementId,
+                Field = issue.Field,
+                Code = issue.Code,
+                Description =
+                    String.IsNullOrEmpty(issue.Description) && !String.IsNullOrEmpty(issue.CustomTextKey)
+                        ? await _translationService.TranslateTextKey(issue.CustomTextKey, language)
+                        : issue.Description,
+                CustomTextKey = issue.CustomTextKey,
+                CustomTextParams = issue.CustomTextParams,
+            })
+        );
+
+        return issuesWithDescription.ToList();
```

Comment:
I think this should be a loop that mutates the issues instead. Translation doesn't typically involve slow http calls, so I don't think trying to do translation in parallel is worth it.